### DISCUSSION TITLE
traefik auth support

### DISCUSSION THREAD

**Discussion by twsl** [logto-io/logto#3098] [Ideas]
I'd really like to use logto to secure my docker containers behind traefik, basically something like authentik with a better UI/UX. What do you think about this?

**Comment by gao-sun** [comment]
thanks @twsl. do you imply to use Logto as an Identity Provider for traefik?

the support of being a IdP is on our roadmap this year. lmk if you meant something else.

**Reply by wuast94** [reply]
https://doc.traefik.io/traefik/middlewares/http/forwardauth/

https://www.authelia.com/reference/guides/proxy-authorization/#forwardauth

here is some more information. the pretty thing about this is you can secure web services that dont have an integrated auth, even ones without any login at all. on the traefik link is a pretty diagram at the top that should make this mway clearer. 

i for example use authentik for this at the moment and would love to switch to logto, but this is a huge blocker for me.

**Comment by waltcow** [comment]
i think it should be used like this
https://github.com/BlackBeltTechnology/traefik-keycloak-sso-reverse-proxy
[traefik-forward-auth ](https://github.com/thomseddon/traefik-forward-auth)

**Comment by wuast94** [comment]
any news on this?

**Comment by twsl** [comment]
I ended up using https://github.com/ItalyPaleAle/traefik-forward-auth

**Reply by iberflow** [reply]
Hi there, with these proxies, does it mean that I'd need to configure the Oauth2 providers (Google, Github, Zendesk, etc) in the proxy AND in logto dashboard? I want to split auth methods between Traefik Dash, Grafana, Logto admin and other services like OpenAPI docs, backoffice admin, etc. Does this mean I would need to run multiple instances of the proxy somehow? Maybe you can share an example of how you deployed it?

**Reply by twsl** [reply]
You just configure everything in logto. Using separate organizations you might be able to setup different login permissions but I haven't looked into that as I am just using it for my homelab

**Comment by demsking** [comment]
Hi everyone 👋

Just to share a small contribution related to this topic — I’ve created a middleware called **Logto Forward Authentication** to help integrate **Logto** with **Traefik** using forward authentication. It acts as a proxy to enforce auth and inject user claims via headers.

- GitLab repo: [https://gitlab.com/demsking/logto-proxy](https://gitlab.com/demsking/logto-proxy)
- Docs: [https://demsking.gitlab.io/logto-proxy](https://demsking.gitlab.io/logto-proxy)

Hope this helps anyone trying to secure services with Logto + Traefik. Feedback and contributions are welcome!
